### PR TITLE
Send millisecond precision timestamps

### PR DESCRIPTION
**Pull request by jaredmixpanel**

### CODE REVIEW

File: lib/mixpanel-node.js [hunk 1]
```diff
@@ -295,13 +291,7 @@ var create_client = function(token, config) {
             properties = {};
         }
 
-        // time is optional for `track` but must be less than 5 days old if set
-        if (properties.time) {
-            properties.time = ensure_timestamp(properties.time);
-            if (properties.time < Date.now() / 1000 - TRACK_AGE_LIMIT) {
```
Comment (jaredmixpanel) [inline lib/mixpanel-node.js:301]: None of our other SDKs perform this check. If we want to keep it we'll need to determine if the provided timestamp is in seconds or milliseconds, but doesn't seem necessary to me.

File: lib/mixpanel-node.js [hunk 2]
```diff
@@ -236,10 +235,7 @@ var create_client = function(token, config) {
                 batch = batch.map(function (event) {
                     var properties = event.properties;
 
-                    if (endpoint === '/import' || event.properties.time) {
-                        // usually there will be a time property, but not required for `/track` endpoint
-                        event.properties.time = ensure_timestamp(event.properties.time);
-                    }
+                    event.properties.time = ensure_timestamp(event.properties.time || Date.now());
```
Comment (tdumitrescu) [inline lib/mixpanel-node.js:238]: Unless I'm misreading, this seems to be a behavior change (here and on L294 below). Previously, if you didn't explicitly set a time for a `track()` call, it wouldn't include the property, and the API server would use the time received as the event timestamp. Now that case will result in the client's clock being used instead, which might be something totally weird for example for an IoT device (one of the ways people use this SDK).

I'd prefer to keep the old way, as it offers the client the flexibility not to include the `time` prop.

File: lib/profile_helpers.js [hunk 3]
```diff
@@ -18,6 +18,8 @@ function merge_modifiers(data, modifiers) {
         }
         if (modifiers.hasOwnProperty("$time")) {
             data.$time = ensure_timestamp(modifiers.$time);
+        } else {
+            data.$time = Date.now();
```
Comment (tdumitrescu) [inline lib/profile_helpers.js:22]: same concern here too

File: lib/mixpanel-node.js (hunk 2 above)
Comment (jaredmixpanel) [reply lib/mixpanel-node.js:238]: You're correct. The change would bring it more in line with the other SDK's but I hadn't considered the IoT-style use-case where you'd prefer our backend timestamp to the clock on some random toaster. I was just thinking you'd prefer your server's timestamp to ours. I'll revert it back.

File: lib/profile_helpers.js (hunk 3 above)
Comment (jaredmixpanel) [reply lib/profile_helpers.js:22]: ... will revert this as well.

**Review by tdumitrescu — APPROVED** [review]
👍

**Pull request merged**